src/App.tsx

```diff
@@ -1,61 +1,174 @@
+import { useState } from 'react';
+import { encode, decode } from 'js-base64';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { ITodo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
 import './App.scss';
+import { IUser } from './types/User';
+
+const preperadTodos = todosFromServer.map(todo => {
+  return {
+    ...todo,
+    user: usersFromServer
+      .find(user => user.id === todo.userId) ?? null,
+  };
+});
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+interface ITodoSetting {
+  userText: string;
+  isUserTextEmpty: boolean;
+  selectedUser: IUser | null;
+  isSubmit: boolean;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState<ITodo[]>(preperadTodos);
+
+  const initialTodoSetting = {
+    userText: '',
+    isUserTextEmpty: true,
+    selectedUser: null,
+    isSubmit: false,
```

Comment:
`isSubmit` sounds really strange.
I'm sure that we can easily refactor code to remove this field

Reply:
Not fixed